cms/grading/languages/python3_cpython.py

```diff
@@ -45,30 +45,25 @@ def source_extensions(self):
         """See Language.source_extensions."""
         return [".py"]
 
+    @property
+    def executable_extension(self):
+        """See Language.executable.extension."""
+        return ".pyz"
+
     def get_compilation_commands(self,
                                  source_filenames, executable_filename,
                                  for_evaluation=True):
         """See Language.get_compilation_commands."""
-        zip_filename = "%s.zip" % executable_filename
-
-        commands = []
-        files_to_package = []
-        commands.append(["/usr/bin/python3", "-m", "compileall", "-b", "."])
-        for idx, source_filename in enumerate(source_filenames):
-            basename = os.path.splitext(os.path.basename(source_filename))[0]
-            pyc_filename = "%s.pyc" % basename
-            # The file with the entry point must be in first position.
-            if idx == 0:
-                commands.append(["/bin/mv", pyc_filename, self.MAIN_FILENAME])
-                files_to_package.append(self.MAIN_FILENAME)
-            else:
-                files_to_package.append(pyc_filename)
-
-        # zip does not support writing to a file without extension.
-        commands.append(["/usr/bin/zip", "-r", zip_filename]
-                        + files_to_package)
-        commands.append(["/bin/mv", zip_filename, executable_filename])
 
+        commands = [["/usr/bin/python3", "-m", "compileall", "-b", "."]]
+
+        # The file with the entry point must be in first position.
+        basename, _ = os.path.splitext(os.path.basename(source_filenames[0]))
+        pyc_filename = "%s.pyc" % basename
+        commands.append(["/bin/mv", pyc_filename, self.MAIN_FILENAME])
+
+        commands.append(["/usr/bin/zip", executable_filename,
+                         self.MAIN_FILENAME] + source_filenames[1:])
```

Comment:
This zips the .py files, not the .pyc - is this intended?

Reply:
Nice catch! Fixed by returning the old implementation.